src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -775,10 +776,27 @@ internal static void CheckParameterModifiers(
 
                     case SyntaxKind.ScopedKeyword when parameterContext is not ParameterContext.FunctionPointer:
                         ModifierUtils.CheckScopedModifierAvailability(parameter, modifier, diagnostics);
-                        Debug.Assert(!seenIn);
-                        Debug.Assert(!seenOut);
-                        Debug.Assert(!seenRef);
-                        Debug.Assert(!seenScoped);
+
+                        if (seenScoped)
+                        {
+                            addERR_DupParamMod(diagnostics, modifier);
+                        }
+                        else if (seenIn || seenOut || seenRef || seenReadonly)
+                        {
+                            // Matches original parsing logic that disallowed parsing out 'scoped' once in/out/ref/readonly had been seen.
+                            diagnostics.Add(ErrorCode.ERR_ScopedAfterInOutRefReadonly, modifier.GetLocation());
+                        }
+                        else if (i < n - 1)
+                        {
+                            // Matches original parsing logic that only allowed 'scoped' to be followed by ref/out/in to
+                            // actually be considered the modifier.
```

Comment:
I didn't understand this comment. The logic below merely reports a diagnostic, how does it make it not considered a modifier?

Reply:
instead of treating it as a modifier (no error), it errors.  meaning we don't accept this as a modifier in this order.

Reply:
updated comment to now say: Only allow 'scoped' followed by `ref/out/in` to actually be considered a valid modifier.